### PR TITLE
execute status callback async

### PR DESCRIPTION
**Pull request by xquanluu**
#392 

Hi @avoylenko Thank you so much for your PR, but I think running status callback notification in asyn, non-blocking mode is much better than current code.

this PR will help us run statusCallback notification in background and don't block the main thread to execute another important tasks.

Could you please help to review this PR as well to see if it's good to merge. It's great if you can test it on your cluster as well.

### CODE REVIEW

**Comment by avoylenko** [comment]
@xquanluu thank you for your help. It looks exactly as I propose in https://github.com/jambonz/jambonz-feature-server/pull/393

File: lib/session/call-session.js [hunk 1]
```diff
@@ -1781,6 +1781,15 @@ class CallSession extends Emitter {
...
+
+    // update calls db
+    //this.logger.debug(`updating redis with ${JSON.stringify(this.callInfo)}`);
+    this.updateCallStatus(Object.assign({}, this.callInfo.toJSON()), this.serviceUrl)
+      .catch((err) => this.logger.error(err, 'redis error'));
+  }
+
+  async executeStatusCallback(callStatus, sipStatus) {
     const {span} = this.rootSpan.startChildSpan(`call-status:${this.callInfo.callStatus}`);
     span.setAttributes(this.callInfo.toJSON());
```
[the remark this answers is not in the document]
Comment (xquanluu) [reply lib/session/call-session.js]: yeap, I agree and update change to PR.
Have you tried to run test with this PR yet?

**Comment by davehorton** [comment]
please note test is failing
```
/home/runner/work/jambonz-feature-server/jambonz-feature-server/lib/session/call-session.js
Error:   1801:7  error  'span' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

```

File: lib/session/call-session.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (xquanluu) [reply lib/session/call-session.js:1794]: actually, the original code is working with that way already.

**Comment by avoylenko** [comment]
Branch [fix/async_status_callback](https://github.com/jambonz/jambonz-feature-server/tree/fix/async_status_callback) tested in my lab. All works as expected. Now its up to @davehorton to approve those changes

**Comment by xquanluu** [comment]
Thanks @avoylenko 
yes I agreed, Could you please also check why testsuite is failing on your branch. Let make sure we don't break legacy behavior.

**Comment by avoylenko** [comment]
@xquanluu those errors are related to my PR, I'm going to close it

**Pull request merged**